my-app/src/app/shared/components/autocomplete-select/autocomplete-select.component.ts

```diff
@@ -0,0 +1,204 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  EventEmitter,
+  forwardRef,
+  Input,
+  OnChanges,
+  OnDestroy,
+  OnInit,
+  Output,
+  SimpleChanges,
+} from '@angular/core';
+import {
+  ControlValueAccessor,
+  FormArray,
+  FormBuilder,
+  FormControl,
+  FormGroup,
+  NG_VALIDATORS,
+  NG_VALUE_ACCESSOR,
+  ValidationErrors,
+  Validator,
+  Validators,
+} from '@angular/forms';
+import { Subject } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
+import { ListOption } from 'src/app/models/list-options.model';
+
+@Component({
+  selector: 'app-autocomplete-select',
+  templateUrl: './autocomplete-select.component.html',
+  styleUrls: ['./autocomplete-select.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  providers: [
+    {
+      provide: NG_VALUE_ACCESSOR,
+      useExisting: forwardRef(() => AutoCompleteSelectComponent),
+      multi: true,
+    },
+    {
+      provide: NG_VALIDATORS,
+      useExisting: forwardRef(() => AutoCompleteSelectComponent),
+      multi: true,
+    },
+  ],
+})
+export class AutoCompleteSelectComponent
+  implements ControlValueAccessor, Validator, OnInit, OnDestroy, OnChanges {
+  @Input() validationMsg: string;
+  @Input() label: string;
+  @Input() set options(options: ListOption[]) {
+    this._options = options
+      ? [...options, ...this._options]
+      : [...this._options];
+  }
+
+  get options(): ListOption[] {
+    return this._options;
+  }
+
+  get OptionsFormArray(): FormArray {
+    return this.formGroup.get('OptionsFormArray') as FormArray;
+  }
+
+  @Input() set isErrors(value: boolean) {
+    this._isErrors = value;
+    if (value) {
+      this._sendCheckedOptions();
+      this._markFormControlsAsTouched(this.OptionsFormArray);
+    }
+  }
+
+  @Input() public placeholder = '';
+  @Input() public selectedValue = '';
+  @Input() public labelField = 'label';
+  @Input() public searchField = this.labelField;
+
+  @Output() public validOptionsList = new EventEmitter<Partial<ListOption[]>>();
+
+  public filteredOptions: string[];
+  public isMatch = true;
+  public isAutocompleteOptionsShown = false;
+  public inputValue = '';
+  public optionsNames: string[] = [];
+
+  public formGroup: FormGroup;
+  private _options: ListOption[] = [];
+  private _destroy$: Subject<void> = new Subject();
+  private _isErrors = false;
+
+  constructor(private fb: FormBuilder) {}
+
+  ngOnInit(): void {
+    this.formGroup = this.fb.group({
+      OptionsFormArray: new FormArray([], [Validators.required]),
+    });
+  }
+
+  writeValue(values): void {
+    if (Boolean(!values?.length)) {
+      return;
+    }
+    this._options = [...values];
+    values.forEach((value) => {
+      this.OptionsFormArray.push(
+        this.fb.control({ value: value.name, disabled: true })
+      );
+    });
+  }
+
+  public registerOnChange(fn: (value: any) => void): void {
+    this.OptionsFormArray.valueChanges
+      .pipe(takeUntil(this._destroy$))
```

Comment:
don't emit when write value is called. Call onChange manually when you select or deselect an author